### PR TITLE
Fix choreo-connect documentation

### PR DESCRIPTION
**Pull request by menakaj**
* Remove unused docs
* Fix caching docs
* Restructure the Overview and Supported features docs
* Add throttling docs
* Add deploying WebSocket API docs

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Deploy WebSocket API in Choreo Connect
+
+A Web Socket API can be deployed in Choreo Connect via API Manager.
+
+Before You begin
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:5]: Before You begin section does not have any steps.

**Review by Mariangela — CHANGES_REQUESTED** [review]

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Deploy WebSocket API in Choreo Connect
+
+A Web Socket API can be deployed in Choreo Connect via API Manager.
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:3]:
```suggestion
A WebSocket API can be deployed in Choreo Connect via API Manager.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Before You begin
+
+### Step 1 - Create a WebSocket API in API Manager
+
+ Please follow [Create a WebSocket API]({{base_path}}/design/create-api/create-streaming-api/create-a-websocket-streaming-api/) and create the WebSocket API.
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:9]:
```suggestion
 For instructions on how to create a WebSocket API, see [Create a WebSocket API]({{base_path}}/design/create-api/create-streaming-api/create-a-websocket-streaming-api/).
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+### Step 2 - Deploy the API in Choreo Connect environment.
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:11]:
```suggestion
### Step 2 - Deploy the API in the Choreo Connect environment
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+Deploy the API in Choreo Connect by following the [Deploy API]({{base_path}}/deploy-and-publish/deploy-on-gateway/deploy-api/deploy-an-api/)
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:12]:
```suggestion
For more information on deploying the API in Choreo Connect, see [Deploy API]({{base_path}}/deploy-and-publish/deploy-on-gateway/deploy-api/deploy-an-api/).
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+### Step 3 - Generate an Access Token to invoke the API
+By default, the WebSocket API is protected by Oauth2 token.
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:15]:
```suggestion
By default, the WebSocket API is protected by an OAuth2 token.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+Generate a JWT Access token by following, [Get a Test Key to Invoke an API]({{base_path}}/consume/invoke-apis/invoke-apis-using-tools/invoke-an-api-using-the-integrated-api-console/#get-a-test-key-to-invoke-an-api)
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:17]:
```suggestion
For more information on generating a JWT Access token, see [Get a Test Key to Invoke an API]({{base_path}}/consume/invoke-apis/invoke-apis-using-tools/invoke-an-api-using-the-integrated-api-console/#get-a-test-key-to-invoke-an-api).
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+### Invoke the API using a WebSocket client
+The websocket api exposed via Choreo Connect, can be invoked by using a WebSocket client.
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:20]:
```suggestion
The WebSocket API exposed via Choreo Connect can be invoked by using a WebSocket client.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+The JWT token should be set as the Authorization header in the initial WebSocket handshake request.
+
+!!!note
+    The same ports 9095 (https) and 9090 (http) are used for WebSocket APIs.
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:24]:
```suggestion
    The same ports 9095 (HTTPS) and 9090 (HTTP) are used for WebSocket APIs.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+Please follow [Step 4 - Invoke the WebSocket API]({{base_path}}/tutorials/streaming-api/create-and-publish-websocket-api/#step-4-invoke-the-websocket-api) on how to invoke a WebSocket API.
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md:26]:
```suggestion
Invoke the WebSocket API by carrying out [Step 4]({{base_path}}/tutorials/streaming-api/create-and-publish-websocket-api/#step-4-invoke-the-websocket-api) in the Create and Publish a WebSocket API tutorial.
```

**Review by Mariangela — APPROVED** [review]

**Review by Mariangela — CHANGES_REQUESTED** [review]
Pls merge my suggestions into the PR and also add the missing prerequisites.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md (hunk 1 above)
Comment (menakaj) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-web-socket-api.md]: Fixed in https://github.com/wso2/docs-apim/pull/3767/commits/03ed857c6384c8187453777bd2b9f1d390ca44d4

**Review by Mariangela — APPROVED** [review]

**Pull request merged**